integration/query_bitwise_test.go

```diff
@@ -15,41 +15,503 @@
 package integration
 
 import (
+	"math"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/options"
 
 	"github.com/FerretDB/FerretDB/integration/shareddata"
 )
 
 func TestQueryBitwiseAllClear(t *testing.T) {
-	t.Skip("TODO https://github.com/FerretDB/FerretDB/issues/442")
+	t.Parallel()
+	ctx, collection := setup(t, shareddata.Scalars)
+
+	// TODO: https://github.com/FerretDB/FerretDB/issues/508
+	_, err := collection.DeleteOne(ctx, bson.D{{"_id", "binary"}})
+	require.NoError(t, err)
+	_, err = collection.DeleteOne(ctx, bson.D{{"_id", "binary-empty"}})
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct {
+		value       any
+		expectedIDs []any
+		err         mongo.CommandError
+	}{
+		"Double": {
+			value: 1.2,
+			err: mongo.CommandError{
+				Code:    9,
+				Name:    "FailedToParse",
+				Message: "Expected an integer: $bitsAllClear: 1.2",
+			},
+		},
+		"DoubleWhole": {
+			value: 2.0,
+			expectedIDs: []any{
+				"double-negative-zero", "double-zero",
+				"int32-min", "int32-zero",
+				"int64-min", "int64-zero",
+			},
+		},
+		"DoubleNegativeValue": {
+			value: float64(-1),
+			err: mongo.CommandError{
+				Code:    9,
+				Name:    "FailedToParse",
+				Message: "Expected a positive number in: $bitsAllClear: -1.0",
+			},
+		},
+
+		"String": {
+			value: "123",
+			err: mongo.CommandError{
+				Code:    2,
+				Name:    "BadValue",
+				Message: "value takes an Array, a number, or a BinData but received: $bitsAllClear: \"123\"",
+			},
+		},
+
+		"Int32": {
+			value: int32(2),
+			expectedIDs: []any{
+				"double-negative-zero", "double-zero",
+				"int32-min", "int32-zero",
+				"int64-min", "int64-zero",
+			},
+		},
+		"Int32NegativeValue": {
+			value: int32(-1),
+			err: mongo.CommandError{
+				Code:    9,
+				Name:    "FailedToParse",
+				Message: "Expected a positive number in: $bitsAllClear: -1",
+			},
+		},
+
+		"MaxInt64": {
+			value: math.MaxInt64,
+			expectedIDs: []any{
+				"double-negative-zero", "double-zero",
+				"int32-zero",
+				"int64-min", "int64-zero",
+			},
+		},
+		"Int64NegativeValue": {
+			value: int64(-1),
+			err: mongo.CommandError{
+				Code:    9,
+				Name:    "FailedToParse",
+				Message: "Expected a positive number in: $bitsAllClear: -1",
+			},
+		},
+
+		"Array": {
```

Comment:
Composite types should be before scalar types: https://pkg.go.dev/github.com/FerretDB/FerretDB@v0.1.1/internal/types#hdr-Mapping

Reply:
Fixed